LibreNMS/OS/Vrp.php

```diff
@@ -298,11 +298,26 @@ public function pollNac()
     {
         $nac = collect();
         // We collect the first table
-        $portAuthSessionEntry = snmpwalk_cache_oid($this->getDeviceArray(), 'hwAccessTable', [], 'HUAWEI-AAA-MIB');
+        $portAuthSessionEntry = snmpwalk_cache_oid($this->getDeviceArray(), 'hwAccessInterface', [], 'HUAWEI-AAA-MIB');
 
         if (! empty($portAuthSessionEntry)) {
-            // If it is not empty, lets add the Extended table
-            $portAuthSessionEntry = snmpwalk_cache_oid($this->getDeviceArray(), 'hwAccessExtTable', $portAuthSessionEntry, 'HUAWEI-AAA-MIB');
+            // If it is not empty, lets add all the necessary OIDs
+            $hwAccessOids = [
+                'hwAccessMACAddress',
+                'hwAccessDomain',
+                'hwAccessUserName',
+                'hwAccessIPAddress',
+                'hwAccessType',
+                'hwAccessAuthorizetype',
+                'hwAccessSessionTimeout',
+                'hwAccessOnlineTime',
+                'hwAccessCurAuthenPlace',
+                'hwAccessAuthtype',
+                'hwAccessVLANID',
+            ];
+            foreach ($hwAccessOids as $hwAccessOid) {
+                $portAuthSessionEntry = snmpwalk_cache_oid($this->getDeviceArray(), $hwAccessOid, $portAuthSessionEntry, 'HUAWEI-AAA-MIB');
```

Comment:
I might be missing something but isn't this just overwriting $portAuthSessionEntry over Again?

Reply:
`snmpwalk_cache_oid` is populating the input array (3rd parameter) and returning the result. So `$portAuthSessionEntry` after the loop filled with all the values.

Reply:
Oh. My bad. Can't we use one of the "multi" functions here instead tho?

Reply:
The multi functions are only for `snmpget` calls. `snmpwalk` does not support multi OIDs.